src/utils/exceptions.ts

```diff
@@ -1,9 +1,9 @@
-import * as ERROR_MSGS from "../constants/error_msgs";
+import * as ERROR_MSGS from '../constants/error_msgs';
 
-export function isStackOverflowExeption(error: Error) {
+export function isStackOverflowExeption(error: unknown | Error): error is RangeError {
```

Comment:
Since `unknown` is a top type, `unknown | Error` is the equivalent of simply `unknown` to the compiler.

Lovely to see the usage of the `is RangeError` type predicate though :+1: 
```suggestion
export function isStackOverflowExeption(error: unknown): error is RangeError {
```

Reply:
I have noticed this situation of using `SomeType | unknown` at least one other time, and I will try to find it, but I know I may miss it.

Reply:
Would `Error | unknown` work better?

Reply:
No, unfortunately it is the same no matter the order @dcavanagh 

Any type union with `unknown` will simply be `unknown` without any of the other union members